File: leetcode/0026. Remove Duplicates from Sorted Array/Solution.swift
// https://leetcode.com/problems/remove-duplicates-from-sorted-array/

class Solution {
    func removeDuplicates(_ nums: inout [Int]) -> Int {
        if nums.count == 0 { return 0 }

        var i = 0
        for j in 1..<nums.count {
            if nums[j] != nums[i] {
                i += 1
                nums[i] = nums[j]
            }
        }

        return i + 1
    }
}

let s = Solution()

var a1 = [1,1,2]
assert(s.removeDuplicates(&a1) == 2)
assert(a1[..<2] == [1,2])

var a2 = [0,0,1,1,1,2,2,3,3,4]
assert(s.removeDuplicates(&a2) == 5)
assert(a2[..<5] == [0,1,2,3,4])

File: leetcode/0026. Remove Duplicates from Sorted Array/Solution.py
# https://leetcode.com/problems/remove-duplicates-from-sorted-array/

from typing import List

class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        if not nums:
            return 0

        i = 0
        for j in range(1, len(nums)):
            if nums[j] != nums[i]:
                i += 1
                nums[i] = nums[j]

        return i + 1

s = Solution()

nums = [1,1,2]
new_l = s.removeDuplicates(nums)
assert new_l == 2
assert nums[:new_l] == [1,2]

nums = [0,0,1,1,1,2,2,3,3,4]
new_l = s.removeDuplicates(nums)
assert new_l == 5
assert nums[:new_l] == [0,1,2,3,4]